var/spack/repos/builtin/packages/stata/package.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import llnl.util.filesystem as fs
+from datetime import datetime
+
+
+class Stata(Package):
+    """STATA is a general-purpose statistical software package developed
+       by StataCorp."""
+
+# Known limitations of this installer:
+# * STATA requires libpng v12 that ships with EL enterprise distros. Spack
+#   provides version 16 with a "depends_on('libpng')" and STATA checks
+#   explicitly for 12. So until 12 is packaged, `yum install libpng`. :-)
+#
+# * This really only installs the command line version of the program. To
+#   install GUI support there are extra packages needed that I can't easily
+#   test right now (should be installable via yum too as a temp workaround):
+#   libgtk-x11-2.0.so libgdk-x11-2.0.so libatk-1.0.so libgdk_pixbuf-2.0.so
```

Comment:
All of these dependencies are packaged in Spack, although I've had a nightmare trying to build them.

Reply:
Pango and gdk-pixbuf are in Spack already. However, I'm not in a position to be able to easily test them at the moment to see if they work with STATA v15. I poked around for a while on the gtk2 libraries and I didn't find in my searching good matches for them in Spack.

I will need the graphical libraries soon, so I think I'm going to leave these for now as "todo's" until I can better test X11 apps with Spack.

Reply:
The `gtkplus` package should provide these. But I'm fine with a TODO.